### ISSUE TITLE
`ord wallet sats` should return all sat ranges not just rare sats

### ISSUE THREAD

**Issue by raphjaph** [ordinals/ord#3792]

**Comment by cryptoni9n** [comment]
Hi Raph - I'm looking at working on this one, but want to ask some questions first.

1) I previously patched the `ord wallet outputs` command on my fork to show the sat ranges per output.  It worked well but I never created a PR for it and think I've since blown away that code.  My question is would it be better to keep `ord wallet sats` as is and just display the sat ranges with the outputs?  That way sats would still only show special sats but outputs would show all?

2) If it is more intuitive to show all the sat ranges in the sats command (either instead of, or in addition to also showing them in outputs), how do you think it should be best displayed?  Today, the special sats get shown like this:

```[
  {
    "sat": 315000000000,
    "output": "227ca0b1b54aac62b31cdb899fcc0103e572cefa27e68d92362f4b1b3ce2b904:0",
    "offset": 0,
    "rarity": "uncommon"
  },
  {
    "sat": 155000000000,
    "output": "1561a009ce7db4199cb77c652796c51dcba22273cecec13356dd9ba4d936d806:0",
    "offset": 0,
    "rarity": "uncommon"
  }
]
````
I suggest something to the effect of:

```[
  {
    "sat": 315000000000,
    "output": "227ca0b1b54aac62b31cdb899fcc0103e572cefa27e68d92362f4b1b3ce2b904:0",
    "range": 315000000000–320000000000,
    "offset": 0,
    "rarity": "uncommon"
  },
  {
    "sat": 155000000000,
    "output": "1561a009ce7db4199cb77c652796c51dcba22273cecec13356dd9ba4d936d806:0",
    "range": 155000000000–160000000000,
    "offset": 0,
    "rarity": "uncommon"
  },
  {
    "output": "b9fa5b9443459832c70e21a67612983bdfbc3e5f3e7c1902e5e896e62a43b351:25",
    "range": 747873459178359–747873459226914, 1457725397711488–1457725398115345, 379295356533988–379295356867957, 106824711233364–106824711413209
   }
]
````

What do you think?

**Comment by mmikrut** [comment]
Hi, perhaps adding  output modifier parameter to `ord wallet sats`  command ( eg: `--uncommon` | `--all`)   would be a good idea?

**Comment by mmikrut** [comment]
I also think that your original patch to add lists of sats to `ord wallet output` is a much desired option, I would love to test it !

Also there is an issue with the format,  if you keep 'rarity'  output you will also need to add special logic to divide sats ranges if there is an uncommon sat in the middle of ranges.  I would drop this line in output,  perhaps leave listing of uncommon one's as it is, its already handled.

**Comment by raphjaph** [comment]
I think `ord wallet sats --all` is a good idea.

The `ord wallet output` improvement as well.

> Also there is an issue with the format, if you keep 'rarity' output you will also need to add special logic to divide sats ranges if there is an uncommon sat in the middle of ranges. I would drop this line in output, perhaps leave listing of uncommon one's as it is, its already handled.

Since rare sats are always the first sat of their range I think this shouldn't be a problem. 

@cryptoni9n Let me know if you want to work on any or all of these features :)

**Comment by cryptoni9n** [comment]
> I think `ord wallet sats --all` is a good idea.
> 
> The `ord wallet output` improvement as well.
> 
> > Also there is an issue with the format, if you keep 'rarity' output you will also need to add special logic to divide sats ranges if there is an uncommon sat in the middle of ranges. I would drop this line in output, perhaps leave listing of uncommon one's as it is, its already handled.
> 
> Since rare sats are always the first sat of their range I think this shouldn't be a problem.
> 
> @cryptoni9n Let me know if you want to work on any or all of these features :)

thanks @raphjaph - will start working on this next.

**Comment by cryptoni9n** [comment]
> I also think that your original patch to add lists of sats to `ord wallet output` is a much desired option, I would love to test it !
merged as #3817
